pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -94,7 +94,7 @@ func (h *handler) onClusterChange(key string, cluster *v3.Cluster) (*v3.Cluster,
 
 // deployK3sBaseUpgradeController creates a rancher k3s/rke2 upgrader controller if one does not exist.
 // Updates k3s upgrader controller if one exists and is not the newest available version.
-func (h *handler) deployK3sBasedUpgradeController(clusterName string) error {
+func (h *handler) deployK3sBasedUpgradeController(clusterName string, isK3s, isRke2 bool) error {
```

Comment:
We're using two bools to set a single value, but there's no possible case where both would be true. Why not create a DistrubutionType enum with values 'k3s' and 'rke2'?

Although I don't even necessarily like that, as it leaks rke2-specific code upstream into k3s. Maybe just pass appName through from an argument to deployK3sBasedUpgradeController, and k3s or rke2 can set is as needed?